cmd/ore/aws/upload.go

```diff
@@ -122,6 +123,26 @@ func defaultBucketURL(urlPrefix, imageName, board, file, region string) (*url.UR
 	return s3URL, nil
 }
 
+type qemuInfo struct {
+	VirtualSize uint64 `json:"virtual-size"`
+}
+
+// We used to hardcode 8GB for Container Linux, but let's support
+// differently-sized images.
+func getVMDKSizeGB(path string) (uint32, error) {
+	out, err := exec.Command("qemu-img", "info", "--output=json", path).Output()
+	if err != nil {
+		return 0, err
+	}
+
+	var info qemuInfo
+	err = json.Unmarshal(out, &info)
+	if err != nil {
+		return 0, err
+	}
+	return uint32(info.VirtualSize / (uint64(1024) * 1024 * 1024)), nil
```

Comment:
This will round down, is that intended?

Reply:
I thought about that, but right now the disk images we're providing are exact multiples of GiB.  We could make it a fatal error to have it not be exact?

I am not clear what the semantics are here - presumably if it wasn't a GiB multiple one would need to round up?

Although now that I actually look at [the docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html) I notice, yeah we'd need to round up but also:

> Default: If you're creating the volume from a snapshot and don't specify a volume size, the default is the snapshot size. 

Which makes sense...so why are we specifying it in the first place?  A little git blame later:

https://github.com/coreos/mantle/commit/f0f298ad37003956ffb9abbae28bb48e956188ed

"The default if nothing is set is 5GB" is confusing - were CL disk images 5GB at that time and he wanted AWS to be larger?

Reply:
CL images are 4.5 GB right now, so this is (a) broken, and (b) a behavior change.

Reply:
Ah.  Well that explains the "5GB" bit probably.  But then where did 8GB come from?

Reply:
We create 8 GB images in AWS, so that the root FS has a reasonable amount of free space.

For compatibility, maybe use max(8 GB, actual image size)?

Reply:
> We create 8 GB images in AWS, so that the root FS has a reasonable amount of free space.

Only for AWS?  Like Azure and other clouds are still 5GB? 

> For compatibility, maybe use max(8 GB, actual image size)?

Hmm, if I do that unconditionally then this entire PR would be pointless since I want to grow RHCOS beyond 8GB, right?

It seems like we'd need a new `--im-not-cl` flag?  Or maybe we only clamp if `amiName.contains("Container-Linux")` ?

Reply:
Sorry, I was confused...yeah, we can do `max`.

Reply:
TFW one finds https://github.com/golang/go/issues/20775

Oh well, these are small integers, we can be confident we're not going to hit FP accuracy issues.